### PR TITLE
WIP: Made Significant API Reference content changes

### PR DESCRIPTION
**Pull request by thomasr8**
Updated existing content and added new sections of the API Reference Guide for API Syntax, Authentication, Rate Limiting, and Summary of API Endpoints.

# Description
In an effort to improve the API Reference documentation, I updated some existing content and added new sections of content to better describe API Syntax, Authentication, Rate Limiting/Throttling, and a Summary of API Endpoints.

# Reasons
I would like to try and improve the API documentation when I see potential changes that could make it easier for API consumers to better understand how to use the CircleCI API.

# Context
A link to a GitHub and/or JIRA issue (if applicable).

### CODE REVIEW

File: jekyll/_api/v1-reference.md [hunk 1]
```diff
@@ -4,72 +4,163 @@ title: CircleCI API v1.1 Reference
...
+  <dt>
+  v1.1 - the API version.
+  </dt>
+</dl>
+
+**Note:** Access to billing functions is only available from the CircleCI application.
+
+### Authentication
+
+The CircleCI API utilizes token-based authentication to manage access to the API server and validate that a user has permission to make API requests. Before you can make an API request, you must first add an API token and then verify that you are authenticated by the API server to make requests. The process to add an API token and have the API server authenticate you is described in the sections below.
```
Comment (ndintenfass) [inline jekyll/_api/v1-reference.md]: Might be worth noting that we also accept cookies, so if you're logged in you can hit these endpoints from the browser.

Comment (felicianotech) [reply jekyll/_api/v1-reference.md]: :+1:

File: jekyll/_api/v1-reference.md [hunk 2]
```diff
@@ -4,72 +4,163 @@ title: CircleCI API v1.1 Reference
...
 
 CircleCI 1.0 and 2.0 are supported by API version `1.1` as documented in the following sections:
 
 * TOC
 {:toc}
 
-## Summary
+### Rate Limiting/Throttling ###
+
+Unlike many other APIs, CircleCI does not currently perform any rate limiting or throttling of API requests to the server, CircleCI may, at its discretion, manually limit the number of requests that can be made to the server if it is determined that the user may be acting inappropriately.
```
[the remark this answers is not in the document]
Comment (felicianotech) [reply jekyll/_api/v1-reference.md]: :+1:

File: jekyll/_api/v1-reference.md [hunk 3]
```diff
@@ -4,72 +4,163 @@ title: CircleCI API v1.1 Reference
...
+    POST - create/add a new API element.
+  </dt> 
+  <dt>
+    PUT - update an existing API element in the API server.
+  </dt>
+  <dt>
+    DELETE - remove/delete an API element in the API server.
+  </dt>
+  <dt>
+</dl>
```
Comment (felicianotech) [inline jekyll/_api/v1-reference.md]: Missing `</dt>` here.

File: jekyll/_api/v1-reference.md [hunk 4]
```diff
@@ -4,72 +4,163 @@ title: CircleCI API v1.1 Reference
...
+
+The CircleCI API consists of the following endpoints, which can be called by making a properly-formatted API request.
 
 <dl>
 <dt>
   GET: /me
 </dt>
 <dd>
-  Provides information about the signed in user.
+  Returns detailed information about the user currently signed into the server.
```
Comment (felicianotech) [inline jekyll/_api/v1-reference.md]: About the user who's creds are being using to contact the API. Either the API token or browser cookie.

File: jekyll/_api/v1-reference.md [hunk 5]
```diff
@@ -81,101 +172,55 @@ All CircleCI API endpoints begin with `"https://circleci.com/api/v1.1/"`.
...
 <dd>
-  Create a new checkout key.
+  Creates a new checkout key.
 </dd>
 <dt>
   GET: /project/:vcs-type/:username/:project/checkout-key/:fingerprint
 </dt>
 <dd>
-  Get a checkout key.
+  Returns an individula checkout key.
```
Comment (felicianotech) [inline jekyll/_api/v1-reference.md]: spelling

**Comment by michelle-luna** [comment]
closing as this was folded into the new API work

**Pull request closed without merging**